woodwork/logical_types.py

```diff
@@ -407,7 +408,12 @@ def _validate_data(self, series):
     def transform(self, series):
         """Validates the series and converts the dtype to match the logical type's if it is different."""
         self._validate_data(series)
-        return super().transform(series)
+
+        typed_ser = super().transform(series)
+        if pdtypes.is_categorical_dtype(typed_ser.dtype):
```

Comment:
Will this check work for Dask and Koalas? Or do we need get the type of Series beforehand?

Reply:
I see that there is a test below for this. Just might be worth considering since we don't do `pdtypes.is_X` anywhere else in **logical_types.py**

Reply:
Yeah, so it used to do that.  However, by the time we call this logic, koalas series have already been cast to the backup type and are not categorical.  So this already kind of handles that case.  We can add the explicit logic back in if you want.

Reply:
Sounds good.